### PR TITLE
[RHELC-1225] Fix an exception when listing third-party packages with Epoch 2

### PR DESCRIPTION
**Pull request by jochapma**
A bug processing output from the repoquery command caused packages with an Epoch equal to 2 to raise an exception in the third-party package listing task. This change fixes the processing.

Jira Issues: [RHELC-1225](https://issues.redhat.com/browse/RHELC-1225)

Checklist

- [x] PR has been tested manually in a VM (either author or reviewer)
- [x] Jira issue has been made public if possible
- [x] `[RHELC-]` is part of the PR title 
- [x] GitHub label has been added to help with Release notes 
- [x] PR title explains the change from the user's point of view
- [x] Code and tests are documented properly
- [x] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

**Review by SpyTec — APPROVED** [review]

**Comment by SpyTec** [comment]
Unit tests are failing but code if works seems good

File: convert2rhel/pkghandler.py [hunk 1]
```diff
@@ -706,8 +706,9 @@ def _get_package_repositories(pkgs):
             repositories_mapping[package] = "N/A"
     else:
         for line in output:
-            if "C2R" in line:
-                split_output = line.lstrip("C2R ").split("&")
+            line = line.lstrip()
```
Comment (SpyTec) [inline convert2rhel/pkghandler.py:709]: I don't think we'll encounter this in the wild tbh, it's just something within our own tests, but doesn't hurt so let's leave it be for now

**Comment by SpyTec** [comment]
Simple and effective
```
>>> pkg = "C2R 2:eog-44.1-1.fc38.x86_64&updates"
>>> pkg.lstrip("C2R ").split("&")
[':eog-44.1-1.fc38.x86_64', 'updates']
>>> pkg[4:].split("&")
['2:eog-44.1-1.fc38.x86_64', 'updates']
```

**Comment by bocekm** [comment]
@jochapma, out of curiosity, why was the code failing with the epoch being specifically 2? Wouldn't other epoch cause the issue as well?

**Comment by jochapma** [comment]
@bocekm, we meant to strip off "C2R" from the left side of the string. Unfortunately, we were using lstrip(), which strips off all instances of any character in the string, so it removed C, 2, and R. So epoch 2 specifically got stripped off.

**Comment by bocekm** [comment]
Ugh, that was a nasty bug indeed, I would have not thought that lstrip() has this behavior. Thanks for the explanation.

**Pull request merged**